src/back/msl.rs

```diff
@@ -403,29 +403,52 @@ impl<W: Write> Writer<W> {
                 }
             }
             crate::Expression::Constant(handle) => {
-                let kind = match module.constants[handle].inner {
+                let width = 32; //TODO: not sure how to get that...
+                match module.constants[handle].inner {
                     crate::ConstantInner::Sint(value) => {
                         write!(self.out, "{}", value)?;
-                        crate::ScalarKind::Sint
+                        Ok(MaybeOwned::Owned(
```

Comment:
We don't need to construct this type and return it from the `match`. We only need to do `write!()` stuff in the match arms. After the match, we could get the type by looking at `module.constants[handle].ty` without guessing.